client/src/App.js

```diff
@@ -6,35 +6,63 @@ import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
 import Home from "./pages/Home";
 import Categories from "./pages/Categories";
 import Notes from "./pages/Notes";
+import Register from "./pages/Register";
+import Login from "./pages/Login";
+import { Provider } from "react-redux";
+import store from "./store";
+import jwt_decode from "jwt-decode";
+import setAuthToken from "./utils/setAuthToken";
+import { setCurrentUser, logoutUser } from "./actions/authActions";
+import PrivateRoutes from "./components/PrivateRoutes";
 import EditNote from "./pages/EditNote";
 
+
+// Check for token to keep user logged in
+if (localStorage.jwtToken) {
+    // Set auth token header auth
+    const token = localStorage.jwtToken;
+    setAuthToken(token);
+    // Decode token and get user info and exp
+    const decoded = jwt_decode(token);
+    // Set user and isAuthenticated
+    store.dispatch(setCurrentUser(decoded));
+    // Check for expired token
+    const currentTime = Date.now() / 1000; // to get in milliseconds
+    if (decoded.exp < currentTime) {
+        // Logout user
+        store.dispatch(logoutUser());
+        // Redirect to login
+        window.location.href = "./login";
+    }
+}
 function App() {
     return (
         <div className="App">
-            <Router>
-                <CssBaseline />
-                <Layout>
-                    <MainContent>
-                        <Switch>
-                            <Route path="/categories/:categoryID">
-                                <Notes />
-                            </Route>
-                            <Route path="/categories">
-                                <Categories />
-                            </Route>
-                            <Route path="/notes/:id">
-                                <EditNote />
-                            </Route>
-                            <Route path="/notes">
-                                <Notes />
-                            </Route>
-                            <Route path="/">
+            <Provider store={store}>
+                <Router>
+                    <CssBaseline />
+                    <Layout>
+                        <MainContent>
+                            <Switch>
                                 <Home />
```

Comment:
Renders `<Home />` by default on every route - is inside the `<Switch>` but outside of any route.
Remove to fix.